src/main/java/rx/internal/operators/OperatorConcat.java

```diff
@@ -187,6 +200,8 @@ void subscribeNext() {
         private final ConcatSubscriber<T> parent;
         private final AtomicInteger once = new AtomicInteger();
```

Comment:
AtomicBoolean is very similar to AtomicInteger since it uses under the hood an integer.
But I agree that AtomicBoolean makes the intent of the code more explicit.